### ISSUE TITLE
Download static file does not work after deploying react app with aws-amplify

### ISSUE THREAD

**Issue by nhhaidee** [aws-amplify/amplify-console#1219]
Hi all,

I have a problem that is download static file does not work after deploying react app with aws-amplify, the size of downloaded file is very small in comparison with actual file. This code works perfectly on localhost. I don't really know what is problem. Can anyone experience this problem like me?

I placed files that I need to download in the public folder 

Please see below is my code
<div className="col-12 mt-5">
                        <Media>
                            <Media body className="ml-5">
                                <h6>Volunteer Application Form</h6>
                                <a href="asset/files/VolunteerApplicationForm_v1.pdf" target="_blank" rel="noreferrer noopener" download="VolunteerApplicationForm.pdf">
                                    <Button>
                                        <i className="fa fa-download"></i> Download Here
                                    </Button>
                                </a>
                            </Media>
                        </Media>
</div> 

Thanks all!

**Comment by nhhaidee** [comment]
<a href="asset/files/VolunteerApplicationForm_v1.pdf" target="_blank" rel="noreferrer noopener" download="VolunteerApplicationForm.pdf">
                                </a>

**Comment by nhhaidee** [comment]
Here is my code:
<a href="asset/files/VolunteerApplicationForm_v1.pdf" 
              target="_blank" rel="noreferrer noopener" 
              download="VolunteerApplicationForm.pdf">
</a>

**Comment by nimacks** [comment]
@nhhaidee can you provide an appId? Is the download triggered by just clicking a link?

**Comment by nhhaidee** [comment]
<a href="asset/files/VolunteerApplicationForm_v1.pdf" target="_blank" rel="noreferrer noopener" download="VolunteerApplicationForm_v1.pdf">

**Comment by nhhaidee** [comment]
### <a href="asset/files/VolunteerApplicationForm_v1.pdf" target="_blank" rel="noreferrer noopener" download="VolunteerApplicationForm_v1.pdf"> ####

**Comment by nhhaidee** [comment]
How can I get appID? I copied my code into comments but not sure why it became https://github.com/aws-amplify/amplify-console/issues/asset/files/VolunteerApplicationForm_v1.pdf

After I deploying with amplify the link is https://www.<my_domain>/asset/files/VolunteerApplicationForm_v1.pdf, I used a tag to implement download this file, but when I click and download it, the actual file size ie very small, I guess there is something with building on Aws-amplify.

There is no problem with localhost: https://localhost:3000/asset/files/VolunteerApplicationForm_v1.pdf

Can you help me on this problem? Thanks

**Comment by nhhaidee** [comment]
I have found the root cause of the problem: in the Rewrites and redirects Rule of my app missing pdf file extension, so I added pdf into source and it works fine.

{
        "source": "</^[^.]+$|\\.(?!(css|pdf|gif|ico|jpg|js|png|txt|svg|woff|ttf|map|json)$)([^.]+$)/>",
        "target": "/index.html",
        "status": "200",
        "condition": null
    }

**Comment by Murphstarr** [comment]
Any suggestions oh what to try if I have configured the rewrite rule and it still does not work on Amplify?

**Comment by JoelRummel** [comment]
@Murphstarr I also configured the rewrite rule and found that Amplify simply wasn't deploying a PDF of mine.

My PDF was in a subfolder within `public`, so my download link was `example.com/resources/my-file.pdf`. This link worked while hosting a build on localhost, but not on Amplify (the PDF simply didn't exist).

I solved it by moving the PDF out of the `resources` subfolder and into the root `public` folder. This changed my download link to just `example.com/my-file.pdf`. Not sure why this fixed it but it's an okay workaround for my use case.
